tests/lib/rules/no-insecure-url.js

```diff
@@ -166,11 +201,48 @@ ruleTester.run(ruleId, rule, {
                     );
                 };
             `,
+            output: `
+                const someSvg: React.FC = () => {
+                    return (
+                        <svg someOtherAttribute="https://ban-example.com/">
+                        </svg>
+                    );
+                };
+            `,
             errors: [
                 { messageId: "doNotUseInsecureUrl", line: 4},
             ],
             parser: testUtils.tsParser,
             parserOptions: testUtils.tsReactParserOptions,
         },
+        {
+            // should fix url correctly
+            code: `var a1 = "http://moz\ti\tlla.org";`,
+            output: `var a1 = ${JSON.stringify("https://moz\ti\tlla.org")};`,
+            errors: [
+                { messageId: "doNotUseInsecureUrl", line: 1},
+            ],
+        },
+        {
+            // should fix url in `` correctly
+            code: "var x1 = `http://foo${multipartExample} http://${multipartExample}.com`;",
+            output: "var x1 = `https://foo${multipartExample} http://${multipartExample}.com`;",
+            errors: [
+                { messageId: "doNotUseInsecureUrl", line: 1},
+            ],
+
+            parserOptions: testUtils.moduleParserOptions
+        },
+        {
+            // should fix url in `` correctly
+            // should fix url correctly
+            code: `var a1 = \`http://moz\ti\tlla.org\`;`,
+            output: `var a1 = \`${JSON.stringify("https://moz\ti\tlla.org").replaceAll('"',"")}\`;`,
```

Comment:
Please write it out explicitly, otherwise it's not clear during review and testing.
Also, I'm not sure this is entirely correct. Please make doubly-sure that it is :)